### PR TITLE
fixes #907 Implement removeIf on HdfsBackedSortedSet

### LINKED ISSUE #907: HdfsBackedSortedSet should override java.util.Collection.removeIf(Predicate)
Calling the removeIf method on HdfsBackedSortedSet results in an exception if the set if persisted.  Just as we do in the remove() method, we should load the underlying FileSortedSet if persisted to enable executing the remove.

### PR DESCRIPTION
**Pull request by ivakegg**
* Implemented removeIf on the supper class BufferedFileBackedSortedSet

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java [hunk 1]
```diff
@@ -476,6 +477,33 @@ public boolean removeAll(Collection<?> c) {
...
+        boolean removed = false;
+        for (SortedSet<E> subSet : set.getSets()) {
+            FileSortedSet<E> fileSet = (FileSortedSet<E>) subSet;
+            if (fileSet.isPersisted()) {
+                try {
+                    fileSet.load();
+                    if (fileSet.removeIf(filter)) {
+                        removed = true;
+                    }
+                    fileSet.persist();
```
Comment (FineAndDandy) [inline warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java:491]: only persist if removed?

**Review by FineAndDandy — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java:491]: nope, I wanted to put it back into a persisted state if it was previously persisted.

Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java:491]: But I get your point, if nothing changed then why rewrite the files.  I will create a separate ticket to add a mechanism to flip from loaded to persisted using the existing underlying files.

Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java:491]: see #949

**Pull request merged**